pkg/gofr/logging/logger.go

```diff
@@ -223,24 +297,69 @@ func colorForGRPCCode(status int32) int {
 }
 
 // NewLogger creates a new logger instance with the specified logging level.
-func NewLogger(level Level) Logger {
+func NewLogger(level Level, args ...interface{}) Logger {
+	var filter Filter
+	if len(args) > 0 {
+		filter = args[0].(Filter)
```

Comment:
This would panic if the args provided do not implement `Filter` interface.
We can do a boolean(ok) check instead.